Libraries/LibWeb/DOM/CharacterData.cpp

```diff
@@ -88,11 +88,15 @@ WebIDL::ExceptionOr<void> CharacterData::replace_data(size_t offset, size_t coun
     // 5. Insert data into node’s data after offset code units.
     // 6. Let delete offset be offset + data’s length.
     // 7. Starting from delete offset code units, remove count code units from node’s data.
-    StringBuilder builder;
-    builder.append(MUST(utf16_view.substring_view(0, offset).to_utf8(Utf16View::AllowInvalidCodeUnits::Yes)));
-    builder.append(data);
-    builder.append(MUST(utf16_view.substring_view(offset + count).to_utf8(Utf16View::AllowInvalidCodeUnits::Yes)));
-    m_data = MUST(builder.to_string());
+    Utf16Data full_data;
```

Comment:
I think you should be able to make this slightly more efficient by preallocating `full_data` before you append anything to it. You can do that like this: `full_data.ensure_capacity(before_data.length_in_code_units() + inserted_data.size() + after_data.length_in_code_units())`. You'll need to move the declarations of `before_data`, `inserted_data` and `after_data` higher up to do this.

Reply:
Should I create a new commit for these changes, or amend the existing commit?

Reply:
You should amend your existing commits when making changes.